File: android/src/main/java/kjd/reactnative/bluetooth/OnReadEventParamWriter.java
package kjd.reactnative.bluetooth;

import android.bluetooth.BluetoothDevice;

import com.facebook.react.bridge.WritableMap;

/**
 * Builds the appropriate WritableMap which is then sent to the React Native Javascript using the
 * {@link com.facebook.react.uimanager.events.RCTEventEmitter}.  This came around due to a request/
 * comment where someone was expecting the onRead Object in a different format.
 */
public interface OnReadEventParamWriter {

    /**
     * Converts the device and message to an appropriate WritableMap.
     *
     * @param device the device from which the message was sent
     * @param contents the String contents of the message
     * @return WritableMap which will be sent to React Native
     */
    public WritableMap write(BluetoothDevice device, String contents);

}

File: ios/OnReadEventParamWriter.swift
//
//  OnReadEventParamWriter.swift
//  RNBluetoothClassic
//
//  Created by Ken Davidson on 2019-10-13.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

/**
  Builds the appropriate Dictionary<String,Any> which is then sent to the React Native.  This came around due to a request/
  comment where someone was expecting the onRead Object in a different format.
 */
protocol OnReadEventParamWriter {
    
    /**
     Converts the device and content to a passable Dictionary<String,Any>
     - parameter device: the device which sent the data
     - parameter data: the data which was received from the device
     */
    func write(device:BluetoothDevice, data:String) -> Dictionary<String,Any>
    
}
